ipykernel/jsonutil.py

```diff
@@ -164,7 +164,10 @@ def json_clean(obj):
         # If all OK, proceed by making the new dict that will be json-safe
         out = {}
         for k,v in iteritems(obj):
-            out[unicode_type(k)] = json_clean(v)
+            if str(type(v)) == "<class 'dict_values'>":
```

Comment:
dict_values can be checked for with:

```python
if PY3:
    from collections.abc import MappingView

...

if PY3 and isinstance(obj, MappingView):
    obj = list(obj)
```

and this should be at the top level, not inside the dict case.